internal/keepers/ocr.go

```diff
@@ -92,9 +119,25 @@ func (k *keepers) Report(ctx context.Context, rt types.ReportTimestamp, _ types.
 	var key [16]byte
 	copy(key[:], hash.Sum(nil))
 
+	// Must not be empty
+	if len(attributed) == 0 {
+		return false, nil, fmt.Errorf("%w: must provide at least 1 observation", ErrNotEnoughInputs)
+	}
+
+	// Build upkeep keys from the given observations
+	upkeepKeys, err := observationsToUpkeepKeys(attributed, k.reportBlockLag)
+	if err != nil {
+		return false, nil, fmt.Errorf("%w: failed to build upkeep keys from the given observations", err)
+	}
+
+	// Check the limit
+	if len(upkeepKeys) > reportKeysLimit {
+		upkeepKeys = upkeepKeys[:reportKeysLimit]
+	}
+
 	// pass the filter to the dedupe function
 	// ensure no locked keys come through
-	keys, err := shuffledDedupedKeyList(attributed, key, k.filter.Filter())
+	keys, err := shuffleUniqueObservations(upkeepKeys, key, k.filter.Filter())
```

Comment:
nit. any reason to rename it? It's still doing the deduping i believe?

Reply:
No I can change it back 👍